src/CMakeLists.txt

```diff
@@ -317,7 +318,9 @@ ELSE()
 	MESSAGE(FATAL_ERROR "Unsupported architecture (CMAKE_SIZEOF_VOID_P is unset)")
 ENDIF()
 
-CONFIGURE_FILE(features.h.in git2/sys/features.h)
+# Generate feature headers
+CONFIGURE_FILE(features.h.in ${libgit2_BINARY_DIR}/include/git2/sys/features.h)
+CONFIGURE_FILE(version.h.in ${libgit2_BINARY_DIR}/include/git2/version.h)
```

Comment:
Generating this file will mean that people _must_ use CMake to build us.  At the moment, at least TortoiseGit uses a build step for us that doesn't involve CMake.

I think that they include us as a submodule.  Then they build us by defining the necessary things (eg `-DGIT_TRACE=1`) directly.  This, and the `git2.rc` changes will definitely break their ability to do this.

What's the motivation for defining the version in cmake instead of `version.h`?

Reply:
The perl bindings also don't use CMake to build the library.

Reply:
@ethomson The motivation is so that we can use it for the `PROJECT()` setup function. I'm not sure whether you're allowed to do things like extracting the version previous to doing the `PROJECT()` call, but we should definitely put the version in there. I'll see whether it's possible to do so with our current `version.h` regex.

Reply:
/cc @csware

Reply:
Forcing users to use CMake isn't a good idea.